internal/runners/packages/packages.go

```diff
@@ -3,49 +3,79 @@ package packages
 import (
 	"strings"
 
-	"github.com/go-openapi/strfmt"
-
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/failures"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/internal/prompt"
 	"github.com/ActiveState/cli/internal/runbits"
 	"github.com/ActiveState/cli/pkg/cmdlets/auth"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
 	"github.com/ActiveState/cli/pkg/platform/model"
 	"github.com/ActiveState/cli/pkg/platform/runtime"
 	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
 )
 
 const latestVersion = "latest"
 
 func executePackageOperation(out output.Outputer, prompt prompt.Prompter, language, name, version string, operation model.Operation) error {
-	fail := auth.RequireAuthentication(locale.T("auth_required_activate"), out, prompt)
-	if fail != nil {
-		return fail.WithDescription("err_activate_auth_required")
+	isHeadless := false
+	if !authentication.Get().Authenticated() {
+		anonymousOk, fail := prompt.Confirm(locale.T("prompt_headless_anonymous"), true)
+		if fail != nil {
+			return locale.WrapInputError(fail.ToError(), "Failed to determine if it is okay to proceed without authorization.")
+		}
+		isHeadless = anonymousOk
+	}
+
+	// Note: User also lands here if answering No to the question about anonymous commit.
+	if !isHeadless {
+		fail := auth.RequireAuthentication(locale.T("auth_required_activate"), out, prompt)
+		if fail != nil {
+			return fail.WithDescription("err_activate_auth_required")
+		}
 	}
 
 	if strings.ToLower(version) == latestVersion {
 		version = ""
 	}
 
 	// Verify that the provided package actually exists (the vcs API doesn't care)
-	var ingredient *model.IngredientAndVersion
+	ingredientNamespace := ""
```

Comment:
There's no situation in which a namespace should ever be empty

Reply:
Hmm... It turns out that we did not need it when I started working on this: https://github.com/ActiveState/cli/blob/9cfa094060a8cb33ea629bbf24b25b49c1d7204e/internal/runners/packages/remove.go#L45

But I think that has actually changed by now, because we have bundles support now.  So, we'll need it for the remove case and then we also needs the `language` again... :) All of this should be fixed now.